### PR TITLE
Fix and Add docstring to nrepl.core

### PR DESCRIPTION
**Pull request by th994**
…mited-transport-seq

Before submitting a PR make sure the following things have been done:

- [ ] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [ ] You've updated the [changelog](../CHANGELOG.md)(that only applies to user-visible changes)

Thanks!

*If you're just starting out to hack on nREPL you might find this [section of its
manual][1] extremely useful.*

[1]: https:/nrepl.org/nrepl/hacking_on_nrepl.html

### CODE REVIEW

**Comment by th994** [comment]
Added and fixed docstring to changes in Pull Request #233.

The docstring in nrepl.core/message was so detailed that it was almost identical to the docstring in nrepl.core/delimited-transport-seq.

Is there any good solution for this?

File: src/clojure/nrepl/core.clj [hunk 1]
```diff
@@ -64,6 +64,14 @@
     (concat head (take 1 tail))))
 
 (defn- delimited-transport-seq
+  "Sends a message via client and returns the head of client's response seq
```
Comment (bbatsov) [inline src/clojure/nrepl/core.clj:67]: Seems to me this returns a function that will do all of the above. The actual `delimited-transport-seq` doesn't do any sending. It's also not returning a seq, which makes its name slightly confusing, but that's not a big deal I guess (and I don't have an idea for a better name).

**Comment by bbatsov** [comment]
Good points. I think the best thing to do here is to differentiate between building the function that does the message sending over the client from the actual process of invoking this function, which results in a message being sent.

File: src/clojure/nrepl/core.clj [hunk 2]
```diff
@@ -64,6 +64,14 @@
...
+   by following the steps described below.
+
+    - Add an :id to the message(argument)
```
Comment (bbatsov) [inline src/clojure/nrepl/core.clj:70]: I guess here it'd be good to clarify that the argument in question is the argument of the function that this function returns. A bit meta, right? :-)

**Comment by th994** [comment]
Thanks for the confirmation and the very clear advice.

I modified the docstring with the following in mind.

- delimited-transport-seq builds a function, and that function will be invoked by nrepl.core/message.
- The 'message' that appears in the docstring of delimited-transport-seq is the argument of the function returned by delimited-transport-seq.

Since delimited-transport-seq is also used in nrepl.core/client-session, excluded specific keys from the docstring such as :id.

Please Let me know if there are any problems

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**